konsent/__main__.py

```diff
@@ -1,688 +1,625 @@
+
+
+# check if not logged in
+def is_not_logged_in(f):
+    @wraps(f)
+    def wrap(*args, **kwargs):
+        if not 'logged_in' in session:
+            return f(*args, **kwargs)
+        else:
+            flash('Du er allerede logget ind', 'danger')
+            return redirect(url_for('index'))
+    return wrap
+
+
+# phase 1, issues
+@app.route('/phase1')
+@is_logged_in
+def phase1():
+
+    update_phases()
+
+    posts = Post.query.filter(
+        Post.union_id == session['connected_union']
+        ).filter(
+        Post.phase == 1
+    ).all()
+
+    if len(posts):
+        return render_template('phase1.html', posts=posts)
+    else:
+        return render_template('phase1.html', msg=NO_RESULTS_ERROR)
+
+
+# phase 2, solution proposals
+@app.route('/phase2')
+@is_logged_in
+def phase2():
+
+    update_phases()
+
+    # find posts
+    posts = Post.query.filter(
+        and_(
+            Post.union_id == session['connected_union'],
+            Post.phase == 2)).all()
+
+    if len(posts):
+        return render_template('phase2.html', posts=posts)
+    else:
+        return render_template('phase2.html', msg=NO_RESULTS_ERROR)
+
+
+# phase 3, solutions
+@app.route('/phase3')
+@is_logged_in
+def phase3():
+
+    update_phases()
+
+    # find posts
+    posts = Post.query.filter(
+        and_(
+            Post.union_id == session['connected_union'],
+            Post.phase == 3,
+            Post.vetoed_by == None
+            )).all()
+
+    if len(posts):
+        return render_template('phase3.html', posts=posts)
+    else:
+        return render_template('phase3.html', msg=NO_RESULTS_ERROR)
+
+
+# single post, phase 1
+@app.route('/phase1/post/<int:post_id>', methods=['GET', 'POST'])
+@is_logged_in
+def post1(post_id):
+    form = UpvoteForm(request.form, meta={'csrf_context': session})
+    post_data = {}
+    # find posts
+    post = Post.query.get(post_id)
+    voted = False
+    # check if user already voted
+    vote = Vote.query.filter(
+        and_(
+            Vote.author_id == session["user_id"],
+            Vote.post == post)
+        ).first()
+    if vote is not None:
+        voted = True
+    post_data['voted'] = voted
+
+    # count total votes
+    post_data['votes'] = len(post.votes)
+
+    # if user submitted a vote request
+    if request.method == 'POST' and form.validate():
+
+        # if user has already voted, remove his vote
+        if post_data['voted']:
+            # decrement vote value
+            post.votes_count -= 1
+            # delete relevant entry in votes
+            db.session.delete(vote)
+
+            # commit to database
+            db.session.add(post)
+            db.session.commit()
+
+            # redirect user
+            return redirect(url_for('phase1'))
+
+        # if user hasnt already voted, count his vote
+        else:
+            # increment vote value
+            post.votes_count += 1
+            # count that this user has now voted
+            vote = Vote()
```

Comment:
i get `TypeError: __init__() missing 2 required positional arguments: 'author' and 'target'` here
any idea?